doc/services/pm/device.rst

```diff
@@ -158,8 +158,47 @@ a particular device. It is important to emphasize that, although the
 state is tracked by the subsystem, it is the responsibility of each device driver
 to handle device actions(:c:enum:`pm_device_action`) which change device state.
 
-Each :c:enum:`pm_device_action` have a direct an unambiguous relationship with
-a :c:enum:`pm_device_state`.
+Each :c:enum:`pm_device_state` state is defined as follows:
```

Comment:
I don't disagree with the contents of the documentation changes, and they are good changes, but why is it happening on this PR? what does it have to do with the pm_device_driver_deinit function being added?

Reply:
There is no wly attempt at anything in this PR :) Consider creating a new PR with only the docs, it will quickly stop making sense as pm_device_driver_deinit() is missing... or you could try rewriting the doc part without the deinit part, and then be stuck forever as atleast I would ask why we are missing this usecase :)

I will instead retitle the PR sine it has indeed changed focus a lot from when it was introduced, docs where added way later as requested by jordan

Reply:
changed title, and discussed PR at seconds arch meeting 20/6/2025